activation/wire/malfeasance_invalid_prev_atx.go

```diff
@@ -0,0 +1,343 @@
+package wire
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"slices"
+
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/signing"
+	"github.com/spacemeshos/go-spacemesh/sql"
+)
+
+//go:generate scalegen
+
+// ProofInvalidPrevAtxV2 is a proof that two distinct ATXs reference the same previous ATX for one of the included
+// identities.
+//
+// We are proving the following:
+//  1. The ATXs have different IDs.
+//  2. Both ATXs have a valid signature.
+//  3. Both ATXs reference the same previous ATX for the same identity.
+//  4. If the signer of one of the two ATXs is not the identity that referenced the same previous ATX, then the included
+//     marriage the included marriage proof is valid.
```

Comment:
Something is off here with the comment :upside_down_face:.

Reply:
Updated